File: src/modifiers/aspectRatio.ts
import { createModifier } from './createModifier';

export interface AspectRatioParams {
  /// Width / height ratio. When omitted, the view's intrinsic ratio is
  /// preserved (only `contentMode` is enforced).
  ratio?: number;
  /// `'fit'` shrinks the view to fit the available space without
  /// cropping; `'fill'` expands to fill, cropping overflow. Default
  /// is `'fit'`.
  contentMode?: 'fit' | 'fill';
}

/// SwiftUI `.aspectRatio(_:contentMode:)`. Generic — works on any view.
export function aspectRatio(params?: AspectRatioParams | number) {
  if (typeof params === 'number') {
    return createModifier('aspectRatio', { ratio: params });
  }
  return createModifier('aspectRatio', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/AspectRatio.swift
import SwiftUI

/// SwiftUI `.aspectRatio(_:contentMode:)`. Generic — works on Image,
/// containers, anything with intrinsic or stretchable dimensions.
enum RNWAspectRatioModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("aspectRatio") { view, params, _ in
            let mode = parseContentMode(params.string("contentMode"))
            if let ratio = params.double("ratio") {
                return AnyView(view.aspectRatio(ratio, contentMode: mode))
            }
            // No ratio → use the source's intrinsic ratio.
            return AnyView(view.aspectRatio(contentMode: mode))
        }
    }

    private static func parseContentMode(_ s: String?) -> ContentMode {
        s == "fill" ? .fill : .fit
    }
}
